test/DebugInfo/debug_compilation_dir.swift

```diff
@@ -0,0 +1,7 @@
+// RUN: %target-swiftc_driver -g \
+// RUN:   -wmo -c -debug-compilation-dir /path/to \
+// RUN:   %s -o - -emit-ir | %FileCheck %s
```

Comment:
This test is good, can we add another one that uses a relative source path too to verify we just have the relative path and debug compilation dir?

Reply:
Sure! This is a nice suggestion.

Reply:
Added a new test file [`debug_compilation_dir_rel.swift`](https://github.com/pofat/swift/pull/1/files#diff-9c65f8c65ec086050984c477a26e2b7e5700333a27b4aed8941bf25171af473f)

Tested it and passed with: 
```shell
${LLVM_PROJECT}/llvm/utils/lit/lit.py -s -vv \
  ${BUILD}/Ninja-RelWithDebInfoAssert/swift-macosx-$(uname -m)/test-macosx-$(uname -m) \
  --filter="debug_compilation_dir_rel.swift"
```